templates/userstyle.tera

```diff
@@ -0,0 +1,275 @@
+  color: {{ crust.hex }} !important;
+}
+
+mark, mark strong {
+  background: {{ yellow.hex }} !important;
+  color: {{ crust.hex }} !important;
+}
+
+/* Highlighted searched item */
+mark[data-markjs] {
+  background-color: {{ lavender.hex }} !important;
+}
+
+/*********************************************************************************
+* PLUGINS
+*********************************************************************************/
+
+/* SPOILERS (https://github.com/martinkorelic/joplin-plugin-spoilers) */
+
+/* Container for spoiler blocks */
+.spoiler-block {
+  border: 0 !important;
+  border-radius: 0px;
+}
+
+/* Container for spoiler title */
+#spoiler-block-title {
+  font-family: var(--font-face) !important;
+  color: {{ text.hex }} !important;
+  background-color: {{ mantle.hex }} !important;
+  border-radius: 0px;
+}
+
+/* Container for spoiler body */
+#spoiler-block-body {
+  font-family: var(--font-face) !important;
+  color: var(--light) !important;
+  background-color: {{ mantle.hex }} !important;
+  border-radius: 0px;
+}
+
+/* Styling of the spoiler block body */
+.summary-content {
+  background-color: {{ mantle.hex }} !important;
+}
+
+.summary-title:before {
+  color: {{ lavender.hex }} !important;
+}
+
+/*********************************************************************************
+* PRINT VERSION
+*********************************************************************************/
+
+@media print {
+
+  #rendered-md, body, th, td {
+    /* background-color: var(--white) !important; */
+    background-color: #ffffff !important;
+    font-family: var(--font-face) !important;
+  }
+
+  p, ul, ol, li {
+    color: var(--black) !important;
+  }
+
+  strong {
+    color: var(--black) !important;
+  }
+  th {
+  border: 1px solid var(--black) !important;
+  color: var(--black) !important;
+  border-bottom: 1px solid var(--black) !important;
+  }
+
+  td {
+  border: 1px solid var(--black) !important;
+  color: var(--black) !important;
+  border-bottom: 1px solid var(--black) !important;
+  }
+
+  h1 {
+    border-bottom: 1px solid var(--black) !important;
+    font-weight: var(--bolder) !important;
+  }
+
+  h2, h3, h4, h5, h6 {
+    border-bottom: 0 !important;
+    font-weight: var(--bolder) !important;
+  }
+
+  hr {
+    border: none;
+    border-bottom: 1px solid var(--black) !important;
+    margin: 2.5em 0 !important;
+  }
+
+  a {
+    color: {{ red.hex }} !important;
+    text-decoration: underline !important;
+  }
+
+/* Inline code */
+.inline-code {
+  background-color: #F8F8F8 !important;
+  border: 0 !important;
+  font-family: var(--font-mono) !important;
+  /* color: {{ yellow.hex }} !important; */
+  font-size: 14px !important;
+}
+
+pre, .hljs {
+  background-color: #F8F8F8 !important;
+  font-family: var(--font-mono) !important;
+  padding: 10px !important;
+  color: {{ crust.hex }} !important;
+  font-size: 14px !important;
+  overflow: scroll !important;
+}
```

Comment:
Looks like Joplin uses Highlight.js, which we have a port for... 🤔 could be worth importing in here?

Reply:
Feel free to do it, but this might require major refactoring. We can't use imported libraries in Joplin usercss iirc so you'd end up hard coding most of the stuff.

Reply:
We can curl it in at build time. Other than that does this PR look fine?